db/db_impl/db_impl.cc

```diff
@@ -1529,9 +1529,13 @@ Status DBImpl::GetImpl(const ReadOptions& read_options,
   }
   if (!done) {
     PERF_TIMER_GUARD(get_from_output_files_time);
+    bool file_above_read_threshold = false;
     sv->current->Get(read_options, lkey, pinnable_val, &s, &merge_context,
                      &max_covering_tombstone_seq, value_found, nullptr, nullptr,
-                     callback, is_blob_index);
+                     callback, is_blob_index, &file_above_read_threshold);
+    if (file_above_read_threshold) {
+      TriggerReadCompactionIfRequired(cfd, sv->current->storage_info());
```

Comment:
Better to move to post processing stage below.

Reply:
Ok, I can do that.